src/session/session.controller.ts

```diff
@@ -179,4 +184,43 @@ export class SessionController {
   ) {
     return await this.sessionService.listOrdersBySession(id);
   }
+
+  @ApiResponse({
+    description: 'Updated session user',
+    type: () => SessionUserUpdateDto,
+    status: HttpStatus.OK,
+  })
+  @HttpCode(HttpStatus.OK)
+  @Patch(':id/updateuser')
+  async updateSessionUser(
+    @Param('id', new ParseMongoIdPipe()) id: Types.ObjectId,
+    @Body() doc: SessionUserUpdateDto,
+  ) {
+    return await this.sessionService.updateSessionUser(id, doc);
+  }
+
+  @ApiResponse({
+    description: 'Get all useable coupon',
+    type: () => CouponDto,
+    status: HttpStatus.OK,
+  })
+  @HttpCode(HttpStatus.OK)
+  @Get(':id/coupons')
+  async getCoupons(@Param('id', new ParseMongoIdPipe()) id: Types.ObjectId) {
```

Comment:
Since we are limiting customers to use only 1 coupon per session.
Why does this endpoint have to be plural?